lib/account.rb

```diff
@@ -0,0 +1,27 @@
+class Account
+  attr_reader :entries
+  attr_reader :balance
+
+  def initialize(entries = [], balance = 2000)
+    @entries = entries
+    @balance = balance
+  end 
+
+  def add_entry(entry)
+    @entries << entry
+  end
+
+  def print_statement
+    puts 'date || credit || debit || balance'
+    @entries.each do |entry|
+      puts "#{entry.date} ||" + " ||" + " #{entry.debit} ||" + " #{balance_update(entry)}" if entry.credit.nil?
```

Comment:
`print_statement` is doing two things at once: generating the statement and updating the balance.  This makes it hard to reason about.